MobileCenter/MobileCenter/Internals/AppDelegate/MSAppDelegateForwarder.m

```diff
@@ -446,29 +403,88 @@ - (void)custom_application:(UIApplication *)application didReceiveRemoteNotifica
 - (void)custom_application:(UIApplication *)application
     didReceiveRemoteNotification:(NSDictionary *)userInfo
           fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
+  __block IMP originalImp = NULL;
+  __block UIBackgroundFetchResult actualFetchResult = UIBackgroundFetchResultNoData;
+  __block short customHandlerCalledCount = 0;
+  __block short customDelegateToCallCount = 0;
+  __block MSCompletionExecutor executors = MSCompletionExecutorNone;
+
+  // This handler will be used by all the delegates, it unifies the results and execte the real handler at the end.
+  void (^commonCompletionHandler)(UIBackgroundFetchResult, MSCompletionExecutor) =
+      ^(UIBackgroundFetchResult fetchResult, MSCompletionExecutor executor) {
+
+        /*
+         * As per the Apple dicumentation:
+         * https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623013-application
+         * The `fetchCompletionHandler` is used to let the app the background time for processing the notification and
+         * download any data that will be displayed to the end users when the app will start again. There can only be
+         * one
+         * `UIBackgroundFetchResult` in the end so there is a need for triage while comparing results from delegates.
+         *
+         * Priorities are:
+         * `UIBackgroundFetchResultNewData`>`UIBackgroundFetchResultFailed`>`UIBackgroundFetchResultNoData`.
+         *  - `UIBackgroundFetchResultNewData` means at least one of the delegates did download something successfully.
+         *  - `UIBackgroundFetchResultFailed` means there was one/several downloads among the delegates but they failed.
+         *  - `UIBackgroundFetchResultNoData` means that none of the delegates had anything to download.
+         */
+        if (fetchResult != actualFetchResult &&
```

Comment:
I don't think you need this `if` statement. With or without the `if` statement the result will be same if you just do `actualFetchResult = fetchResult` and this means you can just use `fetchResult`.
I doubt if it is not working as your expectation.

Reply:
That's right we can remove `fetchResult != actualFetchResult`, it just spares some unnecessary assignments to `actualFetchResult` (that's why I added in the first place) but removing it will probably improve the readiness.